cli/user.go

```diff
@@ -121,18 +120,36 @@ func runSetUser(cmd *cobra.Command, args []string) {
 		mustUsage(cmd)
 		return
 	}
-	hashed, err := security.PromptForPasswordAndHash()
-	if err != nil {
-		log.Error(err)
-		return
+	var err error
+	var hashed []byte
+	switch password {
+	case "":
+		hashed, err = security.PromptForPasswordAndHash()
+		if err != nil {
+			panic(err)
+		}
+	case "-":
+		scanner := bufio.NewScanner(os.Stdin)
+		scanner.Scan()
```

Comment:
I think @tamird has a point though. If you enter `<mypassword><EOF>` then that should be ok, but if you enter `<mypassword><ENTER>someotherstuff<EOF>` then maybe that should return an error. So you'd want `Scan()` to return `false` or error out with a message such as `extra bytes read after newline`. The case in which that could make a difference is if someone is auto-generating passwords and doesn't know newlines aren't permitted (ok, quite exotic, but still).